packages/starknet-snap/src/utils/keyPair.ts

```diff
@@ -43,12 +43,9 @@ function hashKeyWithIndex(key: string, index: number) {
   return number.toBN(hash);
 }
 
-export async function getAddressKey(keyDeriver: BIP44AddressKeyDeriver, addressIndex = 0, useOldAccounts = false) {
+export async function getAddressKey(keyDeriver: BIP44AddressKeyDeriver, addressIndex = 0) {
   const privateKey = (await keyDeriver(addressIndex)).privateKey;
-  const chainCode = (await keyDeriver(addressIndex)).chainCode;
-  const addressKey = useOldAccounts
-    ? `0x${encode.removeHexPrefix(privateKey)}${encode.removeHexPrefix(chainCode)}`
-    : privateKey;
+  const addressKey = privateKey;
```

Comment:
[optional] You may do this instead?
```suggestion
  const addressKey = grindKey(addressKey);
  return {
    addressKey,
    derivationPath: keyDeriver.path,
  };
```